core/src/main/java/greencity/exception/helper/EndpointValidationHelper.java

```diff
@@ -0,0 +1,69 @@
+package greencity.exception.helper;
+
+import greencity.constant.ErrorMessage;
+import greencity.validator.EndpointValidator;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.context.request.WebRequest;
+import java.util.List;
+
+@Component
+public class EndpointValidationHelper {
+    private static final String METHOD_NOT_ALLOWED = "methodNotAllowed";
+    private static final String EXTRA_CHARACTERS = "extraCharacters";
+
+    private EndpointValidationHelper() {
+    }
+
+    public static ResponseEntity<Object> response(HttpRequestMethodNotSupportedException ex, HttpHeaders headers,
+        WebRequest request) {
+        String url = getUrlFromRequest(request);
```

Comment:
you can move it after casting a request to servletWebRequest and get the URL like this
```
servletWebRequest.getRequest().getRequestURL();
```

Reply:
please move it to the` if block`

```
 String method;
 String URL;

        if (request instanceof ServletWebRequest servletWebRequest) {
            HttpServletRequest servletRequest = servletWebRequest.getRequest();
            method = servletRequest.getMethod();
            url = getUrlFromRequest(request);
        } else {
            throw new IllegalArgumentException(ErrorMessage.INVALID_REQUEST_TYPE);
        }
```